examples/swift/hello_world/ViewController.swift

```diff
@@ -38,33 +38,36 @@ final class ViewController: UITableViewController {
   }
 
   private func performRequest() {
-    guard let envoy = self.envoy else {
+    guard let client = self.client else {
       NSLog("Failed to start request - Envoy is not running")
       return
     }
 
     self.requestCount += 1
     NSLog("Starting request to '\(kRequestPath)'")
 
-    let requestID = self.requestCount
     // Note: this request will use an h2 stream for the upstream request.
     // The Objective-C example uses http/1.1. This is done on purpose to test both paths in
     // end-to-end tests in CI.
-    let request = RequestBuilder(method: .get, scheme: kRequestScheme,
-                                 authority: kRequestAuthority,
-                                 path: kRequestPath)
-        .addUpstreamHttpProtocol(.http2)
-        .build()
-    let handler = ResponseHandler()
-      .onHeaders { [weak self] headers, statusCode, _ in
+    let requestID = self.requestCount
+    let headers = RequestHeadersBuilder(method: .get, scheme: kRequestScheme,
+                                        authority: kRequestAuthority, path: kRequestPath)
+      .addUpstreamHttpProtocol(.http2)
+      .build()
+
+    client
+      .newStream()
+      .setOnResponseHeaders { [weak self] headers, _ in
```

Comment:
I sort of prefer the old naming scheme (`onHeaders`) since `setOnResponseHeaders` to a degree suggests the implementation detail that this is a functor captured as a member, whereas I feel `onHeaders` (or `onResponseHeaders`) to me simply suggests "on headers, _do x_". But this is subjective.

Reply:
Discussed this a bit with @junr03 and @buildbreaker when we were first looking at this, and had elected to go this route to avoid confusion between _setting the callbacks_ and _calling the callbacks_. That said, I think `set` might be clearer but don't feel that strongly

Reply:
I get the idea behind distinguishing them, but the user of the public API will never actually be the one calling the callbacks. If anything, I'd be slightly more on board with naming the closure-type something more verbose internally, since it's never really exposed. e.g., `onHeadersCallback`.

Reply:
That's true. However, if we're going for consistency, our other builders expose `addXYZ` and `setXYZ` for our other builders (i.e., `addRetryPolicy`, `set(name:value:)`, so this maintains consistency there. If you feel strongly we can change, but I'd lean towards keeping this for the time being for the sake of consistency